domains/pallets/executor-registry/src/lib.rs

```diff
@@ -556,12 +549,16 @@ mod pallet {
                     "Genesis executor does not have enough balance to stake."
                 );
 
-                Pallet::<T>::apply_register(
-                    &executor,
-                    public_key.clone(),
-                    reward_address,
-                    true,
-                    initial_stake,
+                assert!(
+                    Pallet::<T>::apply_register(
+                        &executor,
+                        public_key.clone(),
+                        reward_address,
+                        true,
+                        initial_stake,
+                    )
+                    .is_ok(),
+                    "Initial total stake weight can not be overflow"
```

Comment:
```suggestion
                    "Initial total stake weight can not overflow"
```

Honestly, I'm unsure `assert` makes more sense than `expect` if it has to be done like this as this message is not straightforward without looking into the internal impl of `apply_register` and it may become inaccurate if `apply_register` adds new potential failures later.

Reply:
either is fine by me. I was suggesting from a consistency PoV